catalog/app/containers/Bucket/Queries/Athena/spec.md

```diff
@@ -0,0 +1,155 @@
+# Athena State Management Specification
+
+## State Types
+
+See type definitions in [`model/utils.ts`](model/utils.ts) and [`model/state.tsx`](model/state.tsx).
+
+```typescript
+type Data<T> = T | undefined | typeof Loading | Error
+type Value<T> = T | undefined | typeof Loading | Error | null
+
+interface DataController<T> {
+  data: Data<T>
+  loadMore: () => void
+}
+
+interface ValueController<T> {
+  value: Value<T>
+  setValue: (v: T | null) => void
+}
+```
+
+**`Data<T>`** - System data from external sources (APIs). Cannot be `null`.
+**`Value<T>`** - User selections and choices. Can be `null` when user deselects.
+
+**DataController** - Manages loading operations, handles `Loading`/`Error` states.
+**ValueController** - Manages user selections, cannot set `Loading`/`Error` directly.
+
+## XML User Stories
+
+### Story 1: View Execution Results
+
+**URL:** `/bucket/{bucket}/athena/workgroup/{workgroupId}/execution/{executionId}`
```

Comment:
what is this URL?

Reply:
from the catalog URL? it doesn't really match any catalog URL.
not sure it really matters, but it feels confusing